### PR TITLE
UCP/TAG-OFFLOAD: add MD index of tag-offload resource

### PR DESCRIPTION
**Pull request by hoopoepg**
- preparation for multi-rail rndv
- added md index of tag offload lane to lower bits

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5413/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5417/ for details (Mellanox internal link).

File: src/ucp/tag/offload.c [hunk 1]
```diff
@@ -428,12 +431,20 @@ ucs_status_t ucp_tag_offload_rndv_zcopy(uct_pending_req_t *self)
...
     size_t iovcnt      = 0;
+    ucp_rsc_index_t md_index;
     ucp_dt_state_t dt_state;
     void *rndv_op;
 
+    md_index = ucp_ep_md_index(ep, req->send.lane);
+
+    UCS_STATIC_ASSERT(ucs_is_pow2(UCS_SYS_CACHE_LINE_SIZE));
+
+    ucs_assert_always(md_index < UCS_SYS_CACHE_LINE_SIZE);
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: can replace this by
```
UCS_STATIC_ASSERT(UCP_MD_INDEX_BITS <= UCS_SYS_CACHE_LINE_SIZE)
ucs_assert(md_index < UCS_SYS_CACHE_LINE_SIZE);
```

File: src/ucp/tag/offload.c [hunk 2]
```diff
@@ -428,12 +431,20 @@ ucs_status_t ucp_tag_offload_rndv_zcopy(uct_pending_req_t *self)
...
+    ucs_assert_always(!((uintptr_t)req & (UCS_SYS_CACHE_LINE_SIZE - 1)));
```
Comment (yosefe) [inline src/ucp/tag/offload.c]: maybe add `#define UCP_TAG_OFFLOAD_REQPTR_MDI_MASK  (UCS_SYS_CACHE_LINE_SIZE - 1)`
and use it in all places, including the assertions?

File: src/ucp/tag/offload.c (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/tag/offload.c]: updated

File: src/ucp/tag/offload.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/tag/offload.c]: done

**Comment by hoopoepg** [comment]
@yosefe squash commits?

**Comment by yosefe** [comment]
yes, please

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5440/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5441/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
@yosefe squashed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5446/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:ornl:retest

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5452/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5453/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5456/ for details (Mellanox internal link).

**Pull request merged**